internal/controller/component/probe_utils.go

```diff
@@ -142,15 +144,30 @@ func buildProbeServiceContainer(component *SynthesizedComponent, container *core
 		"--config", "/config/probe/config.yaml",
 		"--components-path", "/config/probe/components"}
 
-	if len(component.PodSpec.Containers) > 0 && len(component.PodSpec.Containers[0].Ports) > 0 {
+	if len(component.PodSpec.Containers) > 0 {
 		mainContainer := component.PodSpec.Containers[0]
-		port := mainContainer.Ports[0]
-		dbPort := port.ContainerPort
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name:      constant.KBEnvServicePort,
-			Value:     strconv.Itoa(int(dbPort)),
-			ValueFrom: nil,
-		})
+		if len(mainContainer.Ports) > 0 {
+			port := mainContainer.Ports[0]
```

Comment:
This by default behavior may incur bugs and defections in future.

Reply:
This ENV is used to obtain the service port for the database. Since it is directly exposed for ISV configuration without any other rules, this approach is adopted and Service currently follows this strategy as well.